taskgroup.go

```diff
@@ -4,30 +4,44 @@
 // and respond to task errors.
 package taskgroup
 
-import "sync"
+import (
+	"sync"
+	"sync/atomic"
+)
 
 // A Task function is the basic unit of work in a Group. Errors reported by
 // tasks are collected and reported by the group.
 type Task func() error
 
-// A Group manages a collection of cooperating goroutines.  New tasks are added
-// to the group with the Go method.  Call the Wait method to wait for the tasks
-// to complete. A zero value is ready for use, but must not be copied after its
+// A Group manages a collection of cooperating goroutines.  Add new tasks to
+// the group with the Go method.  Call the Wait method to wait for the tasks to
+// complete. A zero value is ready for use, but must not be copied after its
 // first use.
 //
 // The group collects any errors returned by the tasks in the group. The first
 // non-nil error reported by any task (and not otherwise filtered) is returned
 // from the Wait method.
 type Group struct {
-	wg  sync.WaitGroup // counter for active goroutines
-	err error          // error returned from Wait
+	wg      sync.WaitGroup // counter for active goroutines
+	onError ErrorFunc      // called each time a task returns non-nil
 
-	setup sync.Once // set up and start the error collector
-	reset sync.Once // stop the error collector and set err
+	// active is nonzero when the group is "active", meaning there has been at
+	// least one call to Go since the group was created or the last Wait.
+	active atomic.Uint32
 
-	onError func(error) error // called each time a task returns non-nil
-	errc    chan<- error      // errors generated by goroutines
-	edone   chan struct{}     // signals error completion
+	μ   sync.Mutex // guards err
```

Comment:
it also cooperates with active to deliver resettable sync.Once semantics, if you want to call that out here?

Reply:
Can't hurt. 🙂